src/Workspaces/Remote/ServiceHub/Services/KeepAlive/RemoteKeepAliveService.cs

```diff
@@ -15,23 +17,65 @@ protected override IRemoteKeepAliveService CreateService(in ServiceConstructionA
             => new RemoteKeepAliveService(arguments);
     }
 
+    private readonly ConcurrentDictionary<long, TaskCompletionSource<bool>> _sessionIdToCompletionSource = new();
```

Comment:
Nit: add a comment the bool isn't being used, this is just to work around the fact this project compiles with netstandard2.0 and we can't use the non-generic form. I'm also not sure if TaskCompletionSource<object> is better for the JIT, since there's always a risk this has to JIT a specialization here rather than using the System.__Canon one.